azure/Guidelines.md

```diff
@@ -503,9 +509,9 @@ This indicates to client libraries and customers that values of the enumeration
 If you can't avoid them, then follow the guideline below.
 
 <a name="json-use-discriminator-for-polymorphism"></a>
-:white_check_mark: **DO** define a `kind` field indicating the kind of the resource and include any kind-specific fields in the body.
+:white_check_mark: **DO** define a discriminator field indicating the kind of the resource and include any kind-specific fields in the body.
```

Comment:
nit, it is a bit hard for me to understand phrase `kind-specific fields`. Would it be better if we use `"kind"-specific fields` or a bit more detailed like `fields relates to this particular "kind" of the resource`?

Reply:
What do we think of:

```suggestion
:white_check_mark: **DO** define a discriminator field indicating the resource kind in the body, alongside fields that are specific to that kind of resource.
```

Reply:
I'm OK with this wording as it is close enough and most resources have only 1 polymorphic type in them But, in reality, a resource could have multiple polymorphic types in it, for example:
{
   "shapeKind" : "circle",
   "circle" : { ... circle-specific properties... },
   
   "subscriptionKind": "CreditCard",
   "creditCard" : { ... creditCard-specific properties ... },
}

We just need to be aware of this should it come up. I did build a service once many years ago where this was a real issue.

Reply:
I don't believe it is possible to describe a schema with multiple polymorphic types in OpenAPI. A schema can have at most one `discriminator`.